process/block/metablock.go

```diff
@@ -1562,33 +1562,54 @@ func (mp *metaProcessor) ApplyBodyToHeader(hdr data.HeaderHandler, bodyHandler d
 }
 
 func (mp *metaProcessor) createEpochStartForMetablock(metaBlock *block.MetaBlock) (*block.EpochStart, error) {
-	epochStart := &block.EpochStart{
-		LastFinalizedHeaders: make([]block.EpochStartShardData, 0),
+	if !mp.epochStartTrigger.IsEpochStart() {
+		return &block.EpochStart{}, nil
 	}
 
-	if !mp.epochStartTrigger.IsEpochStart() {
-		return epochStart, nil
+	epochStart, lastNotarizedHeaders, err := mp.getLastNotarizedAndFinalizedHeaders()
+	if err != nil {
+		return nil, err
 	}
 
+	pendingMiniBlocks, err := mp.pendingMiniBlocks.PendingMiniBlockHeaders(lastNotarizedHeaders)
```

Comment:
The result of operations between 1574-1584 are not used anywhere

Reply:
it is used. It is written into epochStart.LastFinalizedHeaders[rcvShId].PendingMiniblocks